wordpress-mobile/WordPress-Android
# [Reader] Fix `Reader` showing posts from a language different than the selected one

Fixes #19969 

-----

There are many testing steps, but basically the goal of this PR is to make sure the content shown in the Reader matches the app selected language. If you see any `Reader` content using the wrong language, please let me know.

## To Test:

1. Install Jetpack and sign in.
2. Open "Me" screen.
3. Tap "App Settings".
4. Change the "Interface Language" to a language **different than English and different than the one selected in `wordpress.com`** (it seems like when the language query parameter is missing, the response defaults to the language selected in web. Thanks @thomashorta  for the tip).
5. Open "Reader".
6. Make sure you're following at least one tag and open "Discover" feed.
7. 🔍 Verify the posts list: they should be posts matching the selected language in step 4.
8. Open the search screen (tap on 🔍).
9. 🔍 Search for something: the results should match the selected language in step 4 even if you search use words from a different language (e.g. searching for the word "football" while having Portuguese selected). **Ideally this step should be tested using a version before and after Android O (see [here](https://github.com/wordpress-mobile/WordPress-Android/blob/ac56c495d4f5cecf2a50f349576f23515679d8b1/WordPress/src/main/java/org/wordpress/android/ui/reader/services/search/ReaderSearchServiceStarter.java#L26))**.
10. Open "My Site" tab with a blog selected.
11. Look for the daily prompts card and tap "View all responses".
12. 🔍 The posts should match the selected language in step 4. **Ideally this step should be tested using a version before and after Android O (see [here](https://github.com/wordpress-mobile/WordPress-Android/blob/7843d4e88d4b6ded2656df769e305c9c7f326d0b/WordPress/src/main/java/org/wordpress/android/ui/reader/services/post/ReaderPostServiceStarter.java#L42))**
13. Open "Reader" and select "Discover".
14. Scroll until you find a tags card and select any tag.
15. 🔍 The posts should match the selected language in step 4. **Ideally this step should be tested using a version before and after Android O (see [here](https://github.com/wordpress-mobile/WordPress-Android/blob/7843d4e88d4b6ded2656df769e305c9c7f326d0b/WordPress/src/main/java/org/wordpress/android/ui/reader/services/post/ReaderPostServiceStarter.java#L42))**
16. Unfollow all tags using "Subscriptions" feed.
17. Open "Discover" feed: the empty state should be triggered and you'll see a button (the label in English is "Get Started")
<img width="343" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/e00fb1ee-207f-4166-886b-2884f2b2a64b">


18. 🔍  Tap the button: the suggested tags should match the language selected in step 4.

Feel free to change to pick language at any point: things should work as expected. Just be sure that the selected language is not the same as the one in `wordpress.com`.


-----

## Regression Notes

1. Potential unintended areas of impact

    - Language of "Reader" results

20. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual testing

21. What automated tests I added (or what prevented me from doing so)

    - Had to update legacy code, which is not testable without a significant refactor.

-----

## PR Submission Checklist:


-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):
